### PR TITLE
[Matmul] [Optimization] Disable pipeline pass for amd gpu

### PR DESCRIPTION
**Pull request by binarman**
This PR temporary disables pipeline optimization for amd gpu to fix matmul operation.

related issue #187

### CODE REVIEW

**Comment by binarman** [comment]
I am going to add test in "test_core_amd" a little later

**Comment by binarman** [comment]
@rsanthanam-amd PTAL?

File: python/test/unit/language/test_core_amd.py [hunk 1]
```diff
@@ -1385,6 +1385,81 @@ def kernel_dynamic(out, val, dtype: tl.constexpr):
...
+    a_padded = torch.cat((a, zero_M_K, zero_M_K), 0)
+    a_padded = torch.cat((a_padded, zero_3M_K, zero_3M_K), 1)
+    b_padded = torch.cat((b, zero_K_N, zero_K_N), 0)
+    b_padded = torch.cat((b_padded, zero_3K_N, zero_3K_N), 1)
+    c_padded = torch.matmul(a_padded, b_padded)
+    return c_padded[:SIZE_M, :SIZE_N]
+
+@pytest.mark.parametrize('SIZE_M,SIZE_N,SIZE_K,NUM_WARPS,BLOCK_SIZE_M,BLOCK_SIZE_N,BLOCK_SIZE_K', [
+    # Non-forloop
+#    [64, 32, 64, 4, 64, 32, 64],
```
Comment (rsanthanam-amd) [inline python/test/unit/language/test_core_amd.py:1432]: why are these non-for loop configs commented out?

**Review by rsanthanam-amd — APPROVED** [review]
LGTM.  Need to postpone merge until after the IFU PR has landed.

File: python/test/unit/language/test_core_amd.py (hunk 1 above)
Comment (binarman) [reply python/test/unit/language/test_core_amd.py:1432]: These cases generates only one dot operation, so they do not test anything new, compared to `dot` related tests.

I'll remove them.

File: python/test/unit/language/test_core_amd.py [hunk 2]
```diff
@@ -1385,6 +1385,81 @@ def kernel_dynamic(out, val, dtype: tl.constexpr):
...
+#    [128, 64, 128, 4, 128, 64, 128],
+    # K-Forloop
```
Comment (binarman) [inline python/test/unit/language/test_core_amd.py:1434]:
```suggestion
```

**Comment by rsanthanam-amd** [comment]
@binarman please fix the conflicts.  thanks!

**Pull request merged**